waku/waku_keystore/keystore.nim

```diff
@@ -101,7 +101,15 @@ proc loadAppKeystore*(
   except IOError:
     return err(AppKeystoreError(kind: KeystoreIoError, msg: getCurrentExceptionMsg()))
 
-  return ok(matchingAppKeystore)
+  if matchingAppKeystore.isNone():
```

Comment:
Perhaps this is too generic? Will make it difficult to debug. I would suggest handling each mismatch and being more verbose with the errors:

```
  if data["application"].getStr() != appInfo.application:
    error1
  if data["appIdentifier"].getStr() != appInfo.appIdentifier:
    error2
  if data["version"].getStr() != appInfo.version:
    error3
...
```

Reply:
this is the error log now, thanks:
```
ERR 2024-05-01 18:19:03.684+05:30 App initialization failed                  topics="wakunode main" tid=102289 file=wakunode2.nim:139 error="Failed setting up node: Mounting protocols failed: failed to mount waku RLN relay protocol: failed to mount WakuRlnRelay: could not initialize the group manager: failed to get the keystore credentials: keystore error: JSON value mismatch: Application does not match. Expected 'waku-rln-relay' but got 'nwaku-rln-relay'"
```